core-bundle/src/Resources/contao/drivers/DC_Folder.php

```diff
@@ -2974,7 +2974,7 @@ protected function isValid($strFile)
 			throw new AccessDeniedException('Invalid file name "' . $strFile . '" (hacking attempt).');
 		}
 
-		if (Validator::isInsecurePath($strFolder))
+		if (null === $strFolder || Validator::isInsecurePath($strFolder))
```

Comment:
Doesn’t this change the behaviour? Because `Validator::isInsecurePath(null)` would be `false` I think.

Reply:
We should adjust the `Validator::isInsecurePath()` method instead and return early if `$strPath` is not a string.

Reply:
But in this case we should be careful what we consider to be “secure”. 
IMO every type other than `string|null` should be considered insecure by this method.

Reply:
The method is called `isValid` and should throw an exception if the file is not valid. I would argue that `null` is not a valid file?

Reply:
That‘s true…